client/cache/write_through.go

```diff
@@ -0,0 +1,24 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"github.com/beego/beego/v2/core/berror"
+	"time"
+)
+
+type WriteThoughCache struct {
+	Cache
+	StoreFunc func(ctx context.Context, key string, val any) error
+}
+
+func (w *WriteThoughCache) Set(ctx context.Context, key string, val any, expiration time.Duration) error {
+	if w.StoreFunc == nil {
+		return berror.Error(InvalidStoreFunc, "storeFunc can not be nil")
+	}
```

Comment:
It looks not very elegant. The alternative solutions are:
- remove this validation and we do nothing. Users will get panic when we call StoreFunc. But it's not a big deal because users can easily find the panic if they write any tests. 
- make StoreFunc as private and add one more NewWriteThroughCache function. In this function check if the users pass valid StoreFunc

Reply:
I have chosen the second solution,  I think it is clearer to use